### ISSUE TITLE
A "limit" field in the configuration file is parsed as a `--limit-inference-results` string value

### ISSUE THREAD

**Issue by Mocramis** [PyCQA/pylint#6821]
### Bug description

When "limit" is present in the .pylintrc, it is parsed as the --limit-inference-results.

It was not the case before 2.14.0
(Worked with pylint 2.13.9
astroid 2.11.5
Python 3.10.4 (main, May 14 2022, 05:21:19) [GCC 12.1.0])

### Configuration

```ini
[The name of the section is completely irrelevant !]
limit=8.0
```

### Command used

```shell
# The arguments passed to pylint do not matter witht the faulty pylintrc file 
pylint
```

### Pylint output

```shell
usage: pylint [options]
pylint: error: argument --limit-inference-results: invalid int value: '8.0'
```

### Expected behavior
```shell
# output when running `pylint` with 3.13.9
Usage: pylint [options]

Options:
  -h, --help            show this help message and exit
  --long-help           more verbose help.

  Master:
# ...
```
### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.10.4 (main, May 14 2022, 05:21:19) [GCC 12.1.0]
```

### OS / Environment

Archlinux, linux 5.18.1-arch1-1, Python 3.10.4 
Debian buster, linux 5.4.154, python 3.7.3

**Comment by DanielNoord** [comment]
Could I ask you what the name of your configuration file is?

**Comment by Mocramis** [comment]
.pylintrc in the local directory.

**Comment by DanielNoord** [comment]
Hmm, yeah then this is a somewhat unintended side-effect of our change to `argparse`. We need `argparse` in order to be able to function after Python 3.12. However, `argparse` turns on abbreviations of arguments by default and it behaves differently when this is turned off for some of the earlier version (3.7, 3.8, ...).
So from `2.14` onwards `pylint` options can be abbreviated until there is a class: ``--recursive`` can be ``--re`` as ``--r`` would clash with ``--rcfile``.

This shouldn't be too much of an issue as long as we are not parsing options from other tools. So for `setup.cfg` and `pyproject.toml` we check for `pylint` in section names. For `.pylintrc` we don't as we assume that all options in a file specific to `pylint` are intended for `pylint`. Is there a reason for having `limit` in your `.pylintrc` that you don't want to be parsed by `pylint`? Perhaps we missed an use case when thinking about this.

**Comment by Mocramis** [comment]
Yes, [pylint commit hook conflicts with it](https://github.com/sebdah/git-pylint-commit-hook)

**Comment by DanielNoord** [comment]
> Yes, [pylint commit hook](https://github.com/sebdah/git-pylint-commit-hook) conflicts with it

Ah I did not know about the existence of this package. However, the `limit` options looks a lot like `fail-under` which is baked into `pylint`. I'm going to open an issue on their repository to see if there are any specific reasons why they are not using that.

**Comment by Mocramis** [comment]
Ok, to be honest, I did not use this package anymore but the conf bit did stay around in the pylintrc and i was really surprised about the regression.

Another downside is that understanding a configuration from the documentation (which only documents cmdline arguments) will be even harder now :(

Thanks for your time anyway !

**Comment by DanielNoord** [comment]
> Ok, to be honest, I did not use this package anymore but the conf bit did stay around in the pylintrc and i was really surprised about the regression.
> 
> Another downside is that understanding a configuration from the documentation (which only documents cmdline arguments) will be even harder now :(
> 
> Thanks for your time anyway !

The documentation actually includes information about the configuration files as well! At the bottom of every checker there is an example of how to use those options in a `.toml` file. See for example the small button just above [this section](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#basic-checker).
You can use ``--generate-toml-config`` to generate a ``.toml`` file based on your current configuration as well 😄 

If you have any further suggestions for our documentation please feel free to open a new issue about it! Any concrete and constructive feedback is obviously much appreciated.

I opened https://github.com/sebdah/git-pylint-commit-hook/pull/77 to make the package you mentioned update its `setup.py`. With that I'm going to close this issue, because there is not much more we can do here. It's a downstream package that is causing issues which we can't do much about sadly 😢
